osu.Game/Screens/Play/HUD/FailingLayer.cs

```diff
@@ -31,6 +31,7 @@ public class FailingLayer : HealthDisplay
         /// </summary>
         public double LowHealthThreshold = 0.20f;
 
+        public readonly Bindable<bool> HUDEnabled = new Bindable<bool>();
```

Comment:
Unnecessary. Inject the setting bindable the same way `Player` does.

Reply:
Huh? `HUDOverlay` [does it the same way](https://github.com/ppy/osu/blob/master/osu.Game/Rulesets/Mods/ModBlockFail.cs#L26-L29) I did here.
Or maybe I just didn't get something.

Reply:
you can get the setting bindable by doing
```csharp
config.GetBindable<bool>(OsuSetting.ShowHealthDisplayWhenCantFail);
```
which is a few lines above the lines you linked. there is [an analogous usage is in this same file](https://github.com/ppy/osu/blob/798e8e7a8deea5d1ac665bc9491604c0f082e5ed/osu.Game/Screens/Play/HUD/FailingLayer.cs#L77).

Reply:
I don't think I can, because this bindable isn't just the settings bindable, it's also controlled based on whether or not you have `No Fail` mod ON.